### PR TITLE
Search/replace for line break

### PR DESCRIPTION
**Pull request by sekedus**
*Issue #2869*

*Description of changes:*

* highlight line break
* search line break
* replace line break
ㅤ

Test:
* Open: http://localhost:8888/kitchen-sink.html
* Document: `JavaScript`
* Click: `RegExp` button
* Search/Replace: `\n` or `\n+` or `\n\s+` or `\)\s\{\n\s+`
ㅤ

Auto detect `multiline`, if regexp active and `searchBox` contains line break character `\n`
ㅤ

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by akoreman** [comment]
Thanks for your contribution. Could you expand the description of the PR with a bit more context on what this PR aims to do.

File: src/search.js [hunk 1]
```diff
@@ -318,43 +415,62 @@ var Search = function() {
...
-                        break;
-                    matches.push(m.index, length);
-                }
-                for (var i = matches.length - 1; i >= 0; i -= 2) {
-                    var column = matches[i - 1];
-                    var length = matches[i];
-                    if (callback(row, column, row, column + length))
+                if (multiline) {
+                    var pos = multilineBackward(re, endIndex, row, firstRow);
+                    // console.log(pos);
```
Comment (akoreman) [inline src/search.js:420]: This comment should be removed.

File: src/search_highlight.js [hunk 2]
```diff
@@ -20,22 +20,71 @@ var SearchHighlight = function(regExp, clazz, type) {
...
+
+        return ranges;
+    };
 
     this.update = function(html, markerLayer, session, config) {
         if (!this.regExp)
             return;
         var start = config.firstRow, end = config.lastRow;
+        var multiline = this.isMultilineSearch();
+        console.log(multiline);
```
Comment (akoreman) [inline src/search_highlight.js:71]: This `console.log` should probably not be here.

**Comment by akoreman** [comment]
Closing due to inactivity, feel free to reopen if you want to start working on the PR again.

**Comment by sekedus** [comment]
continues to [#5675](https://github.com/ajaxorg/ace/pull/5675)

**Pull request closed without merging**